content/en/blog/_posts/2023-04-14-1.27-updates-on-speeding-up-pod-startup.md

```diff
@@ -0,0 +1,115 @@
+
+We are introducing a new feature in kubelet that sets a limit on the number of parallel image
+pulls at the node level. This limit restricts the maximum number of images that can be pulled
+simultaneously. If there is an image pull request beyond this limit, it will be blocked until
+one of the ongoing image pulls finishes. Before enabling this feature, please ensure that your
+container runtime's image service can handle parallel image pulls effectively.
+
+To limit the number of simultaneous image pulls, you can configure the `maxParallelImagePulls`
+field in kubelet. By setting `maxParallelImagePulls` to a value of _n_, only _n_ images will
+be pulled concurrently. Any additional image pulls beyond this limit will wait until at least
+one ongoing pull is complete.
+
+You can find more details in the associated KEP: [Kubelet limit of Parallel Image Pulls](https://kep.k8s.io/3673)
+ (KEP-3673). The improvement will arrive as part of Kubernetes v1.27 (scheduled for release in April 2023).
+
+## Raised default API query-per-second limits for kubelet
+
+To improve pod startup in scenarios with multiple pods on a node, particularly sudden scaling
+situations, it is necessary for Kubelet to synchronize the pod status and prepare configmaps,
+secrets, or volumes. This requires a large bandwidth to access kube-apiserver.
+
+In versions prior to v1.27, the default `kubeAPIQPS` was 5 and `kubeAPIBurst` was 10. However,
+Kubelet has increased these defaults to 50 and 100 respectively for better performance during
+pod startup. It's worth noting that this isn't the only reason why we've bumped up the API QPS
+limits for Kubelet.
+
+1. It has a potential to be hugely throttled now (default QPS = 5)
+2. In large clusters they can generate significant load anyway as there are a lot of them
+3. They have a dedicated PriorityLevel and FlowSchema that we can easily control
+
+In old days, we always encounter `volume mount timeout` on kubelet in node with more than 50 pods
+during pod start up. We suggest our customers to bump `kubeAPIQPS` to 20 and `kubeAPIBurst` to 40
+and especially with customers that is using bare-metal(physical machines).
+
+More detials can be found in the KEP <https://kep.k8s.io/1040> and the pull request [#116121](https://github.com/kubernetes/kubernetes/pull/116121).
+
+## Event triggered updates to container status
+
+`Evented PLEG` (PLEG is short for "Pod Lifecycle Event Generator") is set to be in beta for v1.27,
+disabled by default (needs to be enabled explicitly). It aids pod startup by reducing unnecessary
+work during inactivity through replacing periodic polling. The current `Generic PLEG` incurs
+non-negligible overhead due to frequent container status polling, which worsens with Kubelet's
+parallelism, leading to poor performance and reliability issues. For these reasons, it is
+recommended that we use Evented PLEG instead. 
+
+Further details can be found in the KEP <https://kep.k8s.io/3386>.
```

Comment:
Please link to the docs about pod lifecycle change detection (if there aren't any, write some and then link there).